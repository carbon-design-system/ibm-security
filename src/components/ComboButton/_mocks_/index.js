/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
