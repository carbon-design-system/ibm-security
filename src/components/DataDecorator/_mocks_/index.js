/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

const props = {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
const midLine = {
  type: 'MD5',
  value: '0f3deda483df5e5f8043ea20297d243b',
  score: 30,
  href: '#',
};

export default props;

export { props, midLine };
