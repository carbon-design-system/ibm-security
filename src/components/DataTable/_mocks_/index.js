/* eslint-disable react/prop-types */
/**
 * @file Data table mocks.
 * @copyright IBM Security 2019
 */

import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
} from '../../..';

const headers = [
  {
    key: 'lastSighted',
    header: 'Last sighted',
  },
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'type',
    header: 'Type',
  },
  {
    key: 'attackPhase',
    header: 'Attack phase',
  },
  {
    key: 'method',
    header: 'Method',
  },
  {
    key: 'missing',
    header: 'Missing',
  },
];

const isSelectable = false;
const isSortable = false;

const rows = [
  {
    id: '7c34f4507156b35',
    lastSighted: '11/28/2018',
    name: 'IBM X-Force',
    type: 'IPv6',
    attackPhase: 'Initial Compromise',
    method: 'Phishing',
  },
  {
    id: '38347816a51b0c5a8',
    lastSighted: '06/28/2017',
    name: 'AppX',
    type: 'Hash',
    attackPhase: 'Establish Foothold',
    method: 'Deploy backdoor',
  },
  {
    id: '1d6116a2d1d03b4',
    lastSighted: '05/30/2018',
    name: 'Bigfix',
    type: 'URL',
    attackPhase: 'Escalate Privileges',
    method: 'Credential dumping',
  },
  {
    id: 'c52b0419981221765',
    lastSighted: '12/12/2018',
    name: 'Am I Affected',
    type: 'Hash',
    attackPhase: 'Cross-Site Scripting',
    method: 'Malicious Code',
  },
  {
    id: '697626e6f703131d',
    lastSighted: '11/11/2018',
    name: 'QRadar',
    type: 'Hash',
    attackPhase: 'SQL Injection Attack',
    method: 'SQL Vulnerabilities',
  },
  {
    id: 'd568f12fa7b7cac1',
    lastSighted: '12/09/2018',
    name: 'iris',
    type: 'Hash',
    attackPhase: 'Volume Based Attacks',
    method: 'ICMP floods',
  },
  {
    id: '697626e6f70313554',
    lastSighted: '12/10/2018',
    name: 'TISP',
    type: 'Hash',
    attackPhase: 'Application Layer Attacks',
    method: 'GET/POST floods',
  },
  {
    id: '1dd568f12fa7b7cac2',
    lastSighted: '12/11/2018',
    name: 'CLX',
    type: 'Hash',
    attackPhase: 'Protocol Attacks',
    method: 'SYN Floods',
  },
  {
    id: '697626e6f703131',
    lastSighted: '10/12/2018',
    name: 'X-Force Plus',
    type: 'Hash',
    attackPhase: 'Malicious Links',
    method: 'Social Engineering',
  },
  {
    id: 'dd568f12fa7b7cac0',
    lastSighted: '12/16/2018',
    name: 'BigFix & iris',
    type: 'Hash',
    attackPhase: 'Exploit Kits',
    method: 'Scripts',
  },
  {
    id: '697626e6f70313',
    lastSighted: '01/12/2018',
    name: 'Cases',
    type: 'Hash',
    attackPhase: 'Malicious Email',
    method: 'Social Engineering',
  },
];

const page = 1;
const pageSize = 5;
const pageSizes = [5, 10, 20];

/**
 * Render method.
 * @returns {string} The string to render.
 */
const render = ({
  rows,
  headers,
  getHeaderProps,
  getRowProps,
  getTableProps,
  getSelectionProps,
}) => (
  <TableContainer>
    <Table {...getTableProps()}>
      {headers && (
        <TableHead>
          <TableRow>
            <TableSelectAll {...getSelectionProps()} />
            {headers.map(header => (
              <TableHeader
                key={header.id}
                {...getHeaderProps({ header, isSortable })}
              >
                {header.header}
              </TableHeader>
            ))}
          </TableRow>
        </TableHead>
      )}
      <TableBody>
        {rows.map(row => (
          <TableRow {...getRowProps({ row })} key={row.id}>
            <TableSelectRow {...getSelectionProps({ row })} />
            {row.cells.map(cell => (
              <TableCell key={cell.id}>
                {cell.value ? cell.value : 'â€“'}
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </TableContainer>
);

export {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  render,
  rows,
};
