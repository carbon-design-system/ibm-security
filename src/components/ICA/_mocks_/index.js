/**
 * @file Important content area mocks.
 * @copyright IBM Security 2019
 */

const value = 10;

export default {
  label: 'Label',
  value,
  total: value * value,
};
