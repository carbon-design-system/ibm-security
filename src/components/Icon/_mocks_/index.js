/**
 * @file Icon mocks.
 * @copyright IBM Security 2019 - 2021
 */

import { Add16 } from '@carbon/icons-react';

const path = 'M0';
const renderIcon = Add16;

export { path, renderIcon };
