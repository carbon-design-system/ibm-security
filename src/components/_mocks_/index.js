/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
