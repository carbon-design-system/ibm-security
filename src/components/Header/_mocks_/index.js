/**
 * @file Header mocks.
 * @copyright IBM Security 2018
 */

import { url } from '../../_mocks_';
import labels from '../locales/en/Header.json';

const hasSearch = true;
const hasSearchError = false;

const links = {
  product: url(),
  profile: url(),
  registration: url(),
  sign_in: url(),
  sign_out: url(),
  edit_profile: url(),
  notifications_preferences: url(),
  notifications_view_all: url(),
};

const accounts = new Array(3).fill().map((name = `Company`, id) => ({
  id: id.toString(),
  name: `${name} ${id + 1}`,
}));

const notifications = [
  {
    id: `notification__0`,
    description: 'A new request requires approval.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Dec 21 2018 18:57:08 GMT-0600 (Central Standard Time)',
    label: '11min ago',
  },
  {
    id: `notification__1`,
    description: 'There have been 17 failed sign-ins in the last 3 days.',
    href: url(),
    product: 'Application 2',
    datetime: 'Wed May 01 2019 22:22:52 GMT-0500 (Central Daylight Time)',
    label: '1hr ago',
  },
  {
    id: `notification__2`,
    description: 'There are 3 unread messages in the queue.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Jul 06 2018 00:17:49 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__3`,
    description: 'Review the results of the latest scan.',
    href: url(),
    product: 'Application 3',
    datetime: 'Sun Apr 14 2019 20:32:17 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__4`,
    description: 'A new administrator has been added to your account.',
    href: url(),
    product: 'Application 1',
    datetime: 'Sat Oct 20 2018 19:07:04 GMT-0500 (Central Daylight Time)',
    label: '4hrs ago',
  },
  {
    id: `notification__5`,
    description: 'There are 5 items that require your review.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Dec 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '5hrs ago',
  },
  {
    id: `notification__7`,
    description:
      'There are 103 threats found in the last scan. Please click to see the details.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Nov 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '10hrs ago',
  },
  {
    id: `notification__8`,
    description:
      'There are 3 ongoing attacks. Please click to take actions immediately.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Oct 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '24hrs ago',
  },
];

const profile = {
  image_url: null,
  name: {
    first_name: 'Sample',
    surname: 'User',
  },
};

const profileWithAccount = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: accounts[0],
});

const profileWithAccountLongName = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: {
    ...accounts[0],
    name: `${accounts[0].name} plus more text to make the name super super long`,
  },
});
const defaultSearchValue = '127.0.0.1';

export {
  defaultSearchValue,
  hasSearch,
  hasSearchError,
  labels,
  links,
  accounts,
  notifications,
  profile,
  profileWithAccount,
  profileWithAccountLongName,
};
