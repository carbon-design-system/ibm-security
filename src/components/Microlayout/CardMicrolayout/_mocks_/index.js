/**
 * @file Card mocks.
 * @copyright IBM Security 2019
 */

import { icon } from '../../../../../src/components/_mocks_';

const link = '#';

const tag = 'Subtitle';
const title = 'Title';

const text = 'Description';

const image = icon;

const label = 'Image label';

export { image, label, link, tag, text, title };
