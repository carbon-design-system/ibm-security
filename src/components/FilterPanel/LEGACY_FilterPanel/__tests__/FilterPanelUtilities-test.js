/**
 * @file Filter Panel utility tests.
 * @copyright IBM Security 2019
 */

import { filterData } from '../../_mocks_';
import {
  getFilterCategoriesArray,
  getFilterSubcategoriesArray,
  getFiltersArray,
  getSelectedFiltersArray,
} from '../FilterPanelUtilities';

describe('FilterPanelUtilities', () => {
  test('getFilterCategoriesArray gets array of categories', () =>
    expect(getFilterCategoriesArray(filterData)).toMatchSnapshot());

  test('getFilterSubcategoriesArray gets array of subcategories', () =>
    expect(getFilterSubcategoriesArray(filterData)).toMatchSnapshot());

  test('getFiltersArray gets array of filters', () =>
    expect(getFiltersArray(filterData)).toMatchSnapshot());

  test('getSelectedFiltersArray gets array of filters', () => {
    expect(getSelectedFiltersArray(filterData)).toMatchSnapshot();
  });
});
