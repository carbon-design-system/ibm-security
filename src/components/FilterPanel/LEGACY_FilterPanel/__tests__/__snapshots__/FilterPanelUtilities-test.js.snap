// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanelUtilities getFilterCategoriesArray gets array of categories 1`] = `
Array [
  Object {
    "count": 260,
    "id": "DEFAULT",
    "name": "Category",
    "subcategories": Array [
      "DEFAULT",
      "TRUNCATED",
      "NO_FILTERS",
      "ZERO_FILTERS_COUNT",
      "OPEN",
    ],
  },
  Object {
    "count": 0,
    "id": "NO_SUBCATEGORIES",
    "name": "Category with no subcategories",
    "subcategories": Array [],
  },
  Object {
    "count": 0,
    "id": "NO_FILTERS",
    "name": "Category with no filters",
    "subcategories": Array [
      "NO_FILTERS",
    ],
  },
  Object {
    "count": 247,
    "id": "CATEGORY_2",
    "name": "Category 2",
    "subcategories": Array [
      "CATEGORY_2:SUBCATEGORY_1",
      "CATEGORY_2:SUBCATEGORY_2",
      "CATEGORY_2:SUBCATEGORY_3",
      "CATEGORY_2:SUBCATEGORY_4",
      "CATEGORY_2:SUBCATEGORY_5",
      "CATEGORY_2:SUBCATEGORY_6",
    ],
  },
]
`;

exports[`FilterPanelUtilities getFilterSubcategoriesArray gets array of subcategories 1`] = `
Array [
  Object {
    "count": 30,
    "filters": Array [
      "DEFAULT",
      "LONG_TEXT",
      "SELECTED",
    ],
    "id": "DEFAULT",
    "name": "Subcategory",
    "parent": "DEFAULT",
  },
  Object {
    "count": 200,
    "filters": Array [
      "TRUNCATED:FILTER_1",
      "TRUNCATED:FILTER_2",
      "TRUNCATED:FILTER_3",
      "TRUNCATED:FILTER_4",
      "TRUNCATED:FILTER_5",
      "TRUNCATED:FILTER_6",
      "TRUNCATED:FILTER_7",
      "TRUNCATED:FILTER_8",
      "TRUNCATED:FILTER_9",
      "TRUNCATED:FILTER_10",
      "TRUNCATED:FILTER_11",
      "TRUNCATED:FILTER_12",
      "TRUNCATED:FILTER_13",
      "TRUNCATED:FILTER_14",
      "TRUNCATED:FILTER_15",
      "TRUNCATED:FILTER_16",
      "TRUNCATED:FILTER_17",
      "TRUNCATED:FILTER_18",
      "TRUNCATED:FILTER_19",
      "TRUNCATED:FILTER_20",
    ],
    "id": "TRUNCATED",
    "name": "Truncated subcategory",
    "parent": "DEFAULT",
  },
  Object {
    "count": 0,
    "filters": Array [],
    "id": "NO_FILTERS",
    "name": "Subcategory with no filters",
    "parent": "DEFAULT",
  },
  Object {
    "count": 0,
    "filters": Array [
      "ZERO_FILTERS_COUNT:FILTER_1",
      "ZERO_FILTERS_COUNT:FILTER_2",
      "ZERO_FILTERS_COUNT:FILTER_3",
    ],
    "id": "ZERO_FILTERS_COUNT",
    "name": "Subcategory with filters with no count",
    "parent": "DEFAULT",
  },
  Object {
    "count": 30,
    "filters": Array [
      "OPEN:FILTER_1",
      "OPEN:FILTER_2",
      "OPEN:FILTER_3",
    ],
    "id": "OPEN",
    "name": "Open subcategory",
    "open": true,
    "parent": "DEFAULT",
  },
  Object {
    "count": 64,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_1:FILTER_1",
      "CATEGORY_2:SUBCATEGORY_1:FILTER_2",
      "CATEGORY_2:SUBCATEGORY_1:FILTER_3",
      "CATEGORY_2:SUBCATEGORY_1:FILTER_4",
      "CATEGORY_2:SUBCATEGORY_1:FILTER_5",
      "CATEGORY_2:SUBCATEGORY_1:FILTER_6",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_1",
    "name": "Subcategory 1",
    "parent": "CATEGORY_2",
  },
  Object {
    "count": 15,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_2:FILTER_1",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_2",
    "name": "Subcategory 2",
    "parent": "CATEGORY_2",
  },
  Object {
    "count": 25,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_3:FILTER_1",
      "CATEGORY_2:SUBCATEGORY_3:FILTER_2",
      "CATEGORY_2:SUBCATEGORY_3:FILTER_3",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_3",
    "name": "Subcategory 3",
    "parent": "CATEGORY_2",
  },
  Object {
    "count": 60,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_4:FILTER_1",
      "CATEGORY_2:SUBCATEGORY_4:FILTER_2",
      "CATEGORY_2:SUBCATEGORY_4:FILTER_3",
      "CATEGORY_2:SUBCATEGORY_4:FILTER_4",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_4",
    "name": "Subcategory 4",
    "parent": "CATEGORY_2",
  },
  Object {
    "count": 32,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_5:FILTER_1",
      "CATEGORY_2:SUBCATEGORY_5:FILTER_2",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_5",
    "name": "Subcategory 5",
    "parent": "CATEGORY_2",
  },
  Object {
    "count": 51,
    "filters": Array [
      "CATEGORY_2:SUBCATEGORY_6:FILTER_1",
      "CATEGORY_2:SUBCATEGORY_6:FILTER_2",
      "CATEGORY_2:SUBCATEGORY_6:FILTER_3",
    ],
    "id": "CATEGORY_2:SUBCATEGORY_6",
    "name": "Subcategory 6",
    "parent": "CATEGORY_2",
  },
]
`;

exports[`FilterPanelUtilities getFiltersArray gets array of filters 1`] = `
Array [
  Object {
    "count": 10,
    "enabled": false,
    "id": "DEFAULT",
    "name": "Filter",
    "parent": "DEFAULT",
  },
  Object {
    "count": 10,
    "enabled": true,
    "id": "SELECTED",
    "name": "Selected filter",
    "parent": "DEFAULT",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "LONG_TEXT",
    "name": "Long filter label that overflows out of panel.",
    "parent": "DEFAULT",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_1",
    "name": "Filter 1",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_2",
    "name": "Filter 2",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_3",
    "name": "Filter 3",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_4",
    "name": "Filter 4",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_5",
    "name": "Filter 5",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_6",
    "name": "Filter 6",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_7",
    "name": "Filter 7",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_8",
    "name": "Filter 8",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_9",
    "name": "Filter 9",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_10",
    "name": "Filter 10",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_11",
    "name": "Filter 11",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_12",
    "name": "Filter 12",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_13",
    "name": "Filter 13",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_14",
    "name": "Filter 14",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_15",
    "name": "Filter 15",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_16",
    "name": "Filter 16",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_17",
    "name": "Filter 17",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_18",
    "name": "Filter 18",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_19",
    "name": "Filter 19",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "TRUNCATED:FILTER_20",
    "name": "Filter 20",
    "parent": "TRUNCATED",
  },
  Object {
    "count": 0,
    "enabled": false,
    "id": "ZERO_FILTERS_COUNT:FILTER_1",
    "name": "Filter 1",
    "parent": "ZERO_FILTERS_COUNT",
  },
  Object {
    "count": 0,
    "enabled": false,
    "id": "ZERO_FILTERS_COUNT:FILTER_2",
    "name": "Filter 2",
    "parent": "ZERO_FILTERS_COUNT",
  },
  Object {
    "count": 0,
    "enabled": false,
    "id": "ZERO_FILTERS_COUNT:FILTER_3",
    "name": "Filter 3",
    "parent": "ZERO_FILTERS_COUNT",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "OPEN:FILTER_1",
    "name": "Filter 1",
    "parent": "OPEN",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "OPEN:FILTER_2",
    "name": "Filter 2",
    "parent": "OPEN",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "OPEN:FILTER_3",
    "name": "Filter 3",
    "parent": "OPEN",
  },
  Object {
    "count": 5,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 12,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_2",
    "name": "Filter 2",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 19,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_3",
    "name": "Filter 3",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 22,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_4",
    "name": "Filter 4",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 5,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_5",
    "name": "Filter 5",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 1,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_6",
    "name": "Filter 6",
    "parent": "CATEGORY_2:SUBCATEGORY_1",
  },
  Object {
    "count": 15,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_2:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_2",
  },
  Object {
    "count": 13,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_3:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_3",
  },
  Object {
    "count": 3,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_3:FILTER_2",
    "name": "Filter 2",
    "parent": "CATEGORY_2:SUBCATEGORY_3",
  },
  Object {
    "count": 9,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_3:FILTER_3",
    "name": "Filter 3",
    "parent": "CATEGORY_2:SUBCATEGORY_3",
  },
  Object {
    "count": 26,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_4",
  },
  Object {
    "count": 4,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_2",
    "name": "Filter 2",
    "parent": "CATEGORY_2:SUBCATEGORY_4",
  },
  Object {
    "count": 27,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_3",
    "name": "Filter 3",
    "parent": "CATEGORY_2:SUBCATEGORY_4",
  },
  Object {
    "count": 3,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_4",
    "name": "Filter 4",
    "parent": "CATEGORY_2:SUBCATEGORY_4",
  },
  Object {
    "count": 10,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_5:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_5",
  },
  Object {
    "count": 22,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_5:FILTER_2",
    "name": "Filter 2",
    "parent": "CATEGORY_2:SUBCATEGORY_5",
  },
  Object {
    "count": 24,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_6:FILTER_1",
    "name": "Filter 1",
    "parent": "CATEGORY_2:SUBCATEGORY_6",
  },
  Object {
    "count": 7,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_6:FILTER_2",
    "name": "Filter 2",
    "parent": "CATEGORY_2:SUBCATEGORY_6",
  },
  Object {
    "count": 20,
    "enabled": false,
    "id": "CATEGORY_2:SUBCATEGORY_6:FILTER_3",
    "name": "Filter 3",
    "parent": "CATEGORY_2:SUBCATEGORY_6",
  },
]
`;

exports[`FilterPanelUtilities getSelectedFiltersArray gets array of filters 1`] = `
Array [
  Object {
    "count": 10,
    "enabled": true,
    "id": "SELECTED",
    "name": "Selected filter",
    "parent": "DEFAULT",
  },
]
`;
